server/lib/resolvers/wrapResolverWithHooks.js

```diff
@@ -0,0 +1,36 @@
+const _ = require('lodash')
+const {log} = require('../util/logger')
+const request = require('request')
+
+exports.wrapResolverWithHooks = function wrapResolverWithHooks (resolver, resolverMapping) {
+  return wrapResolverWithMappingHooks(resolver, resolverMapping)
+}
+
+function wrapResolverWithMappingHooks (resolverFn, resolverMapping) {
+  return (obj, args, context, info) => {
+    return new Promise(async (resolve, reject) => {
+      if (resolverMapping.preHook && !_.isEmpty(resolverMapping.preHook)) {
+        log.info(resolverMapping.preHook)
+        try {
+          request.get(resolverMapping.preHook)
```

Comment:
I know it seems like a minor thing right now but I would suggest using a Promise based http lib such as [Axios](https://github.com/axios/axios) instead of `request`.

The reason is because request uses older conventions that are not considered good practice any more (callbacks with 3 args), which makes it impossible to use with features like async/await.

Right now it makes no difference to the code so you might ask why would I care that much? If we ever need to add any other http related functionality later (such as HTTP resolvers) the person who implements that will be inclined to use whichever library gets added as part of this PR. request will cause problems with code maintainability in the future.

Reply:
that's a good point. Let's use a promise based lib even thought we don't need it right now.